ui/raidboss/emulator/ui/EmulatedPartyInfo.js

```diff
@@ -219,31 +238,40 @@ export default class EmulatedPartyInfo extends EventBus {
     const $initDataViewer = this.$jsonViewerTemplate.cloneNode(true);
     $initDataViewer.textContent = JSON.stringify(per.initialData, null, 2);
 
-    $container.append(this._wrapCollapse('Initial Data', $initDataViewer, () => {
-      $initDataViewer.textContent = JSON.stringify(per.initialData, null, 2);
-    }));
+    $container.append(this._wrapCollapse(
+        '00:00',
+        'Initial Data',
+        null,
+        null,
+        ['data'],
+        $initDataViewer, () => {
+          $initDataViewer.textContent = JSON.stringify(per.initialData, null, 2);
+        }));
 
     const $triggerContainer = $container.querySelector('.d-flex.flex-column');
 
     for (const i in per.triggers.sort((l, r) => l.resolvedOffset - r.resolvedOffset)) {
       const $triggerDataViewer = this.$jsonViewerTemplate.cloneNode(true);
-      const buttonName = this.getTriggerFiredLabelTime(per.triggers[i]) +
-        ' - ' + per.triggers[i].triggerHelper.trigger.id;
-      const $trigger = this._wrapCollapse(buttonName, $triggerDataViewer, () => {
-        $triggerDataViewer.textContent = JSON.stringify(per.triggers[i], null, 2);
-      });
-      const $buttonWrapper = $trigger.querySelector('.wrap-collapse-button');
-      const $label = this.$triggerLabelTemplate.cloneNode(true);
-      const $labelText = $label.querySelector('.trigger-label-text');
-      const $labelTime = $label.querySelector('.trigger-label-time');
-      $labelText.textContent = this.getTriggerLabelText(per.triggers[i]);
-      $labelTime.textContent = this.getTriggerResolvedLabelTime(per.triggers[i]);
-      $buttonWrapper.append($label);
+      const triggerText = this.getTriggerLabelText(per.triggers[i]);
+      const $trigger = this._wrapCollapse(
+          this.getTriggerResolvedLabelTime(per.triggers[i]),
+          per.triggers[i].triggerHelper.trigger.id,
+          this.getTriggerLabelIcon(per.triggers[i]),
+          triggerText,
+          [per.triggers[i].status.responseType],
+          $triggerDataViewer, () => {
+            $triggerDataViewer.textContent = JSON.stringify(per.triggers[i], null, 2);
+          });
       if (per.triggers[i].status.executed)
         $trigger.classList.add('trigger-executed');
       else
         $trigger.classList.add('trigger-not-executed');
 
+      if (triggerText === undefined || triggerText === null)
```

Comment:
This is a case of defensive programming due to raidemulator being in JS still. I can drop the check against undefined in this case but I think there are some edge cases where `PopupText`'s `valueOrFunction` can return `undefined`, or at least I recall running into a few cases where that happened with my own custom triggers.

Reply:
Oh yes, `valueOrFunction` can return undefined.  I think it will return any type but object or function.  But it all goes through `getTriggerLabelText`, so my thought was that function can fix anything up to be string or null, rather than making everything else deal with that.

Reply:
Switched over to defaulting to undefined since it makes handling the object-as-parameter of wrapCollapse a bit more clean. Cleaned this up otherwise.